File: src/frontend/libs/data-access/src/lib/models/reservation/customer-details.dto.ts
/**
 * Customer details
 */
import type { ISODateString, EmailAddress, PhoneNumber, FirstName, LastName } from '../common';

export type CustomerDetails = {
  readonly firstName: FirstName;
  readonly lastName: LastName;
  readonly email: EmailAddress;
  readonly phoneNumber: PhoneNumber;
  readonly dateOfBirth: ISODateString;
};

File: src/backend/Services/Reservations/OrangeCarRental.Reservations.Api/Shared/CustomerDetailsDto.cs
namespace SmartSolutionsLab.OrangeCarRental.Reservations.Api.Shared;

/// <summary>
///     Customer details for inline registration.
/// </summary>
public sealed record CustomerDetailsDto(
    string FirstName,
    string LastName,
    string Email,
    string PhoneNumber,
    DateOnly DateOfBirth);
